### DISCUSSION TITLE
New Mod Idea : Finger Control

### DISCUSSION THREAD

**Discussion by LanTenDragon** [ppy/osu#13064] [Ideas]
**Describe the new feature:** A new multimod: Finger Control & Finger Control Strict

**Proposal designs of the feature:**

* Finger Control (FC): subtracts from the current score depending on how many extra presses

* Finger Control Strict (FS): Instant fail if key presses at this moment is more than the amount of hitobjects that have appeared up to this point. Note that player is allowed to have less than or equal keypresses, but not more.
  
* This mod should be a **difficulty increase** mod, as you are restricting yourself to a limited amount of keypresses. This mod should be incompatible with Sudden Death, Perfect. 

* This mod is not really an anti-mashing mechanism per se, its just an artificial handicap to showcase one's finger control ability. Plus, with the fail criteria, it would not prevent Hidamari no Uta esque double tapping.  

**Topics needing further discussion:**
* Acronym of FC conflicts with Full Combo (also called FC in rhythm game jargon)
* Should key presses before the first hit object be ignored?
* How would 2B mapping interfere with this mod?
* Should spinners be taken into account?
* If there are anti-mashing mechanisms planned, then this mod is not needed.
* How much score multiplier to give.

**Comment by 59xa** [comment]
I have a question for this. Will this mod be useable across different game modes?

**Comment by bdach** [comment]
It appears to me that this mod would only make sense in the case where there are distinguished keyboard inputs you need to press to the beat (by which I mean taiko and mania), since these are the ones you might try to press "too many keys" on and still get hits, and not very much sense in standard (where a hit is a hit) or in catch (not sure how it would have to even be interpreted there). But I see no reason for it to be impossible to reuse the same mod in taiko and mania if it is implemented in a way that enables it.

**Comment by 59xa** [comment]
If this was to be implemented for catch mode, it will be based on the amount of transitions (fruit change) you need to do without exceeding the maximum limit on a specific beatmap, correct?

**Comment by bdach** [comment]
I suppose it could be done by calculating the minimum amount of left-right input switches autoplay does to pass, true.

**Comment by 59xa** [comment]
So which proposal are you in favour of based on discussion, this issue or #5686? I would like to know which one is a better proposal than the other.

**Comment by bdach** [comment]
Personally I don't feel comfortable making that sort of decision, I just mostly code here. Going to defer to the core team and/or opinions of players if they are around to give some on this matter.

**Comment by LanTenDragon** [comment]
With this being discussed further recently, I realised my proposal lacked some extra details.

>Will this mod be useable across different game modes?

@dscnnctd Yes I intend it to be implemented in the four main rulesets. For future custom rulesets, it is up to the ruleset author to implement.

>PP was not involved in the punishment?

@genericthing I intended score penalty as punishment, as I had lazer's scoring system in mind. Lazer seems to bring 1 million scoring into focus and potentially shift away from PP. But I do agree that if you FC any rhythmically complex beatmap with this mod turned on, then you should be rewarded a healthy amount of PP.

>Not an anti-mashing mechanism

I really forgot to add this part in. I was speaking specifically about being unable to prevent standards ruleset's mashing. That's why I mentioned ```Hidamari No Uta```. 

To expand on what I meant, its not fully an anti-mashing mechanism, but rather it is the intended by-product of finger control mod

**Comment by LanTenDragon** [comment]
>not very much sense in standard (where a hit is a hit)

@bdach There are situations in standard mode where players would do (for a lack of a better word) an empty press. Check the keypress counter at the end of these comparison video.

https://www.youtube.com/watch?v=Qyo9coEuj2k   
https://www.youtube.com/watch?v=OaOBsbwZ7SM

Out of four plays accross two maps, it seems only one had `keypresses == number of hitobjects`

Looks like this would become a really difficult mod on maps with complex rhythms, especially with strict mode.

**Comment by cl8n** [comment]
was going to suggest a very similar mod for mania and found this thread. I wanted a way to penalize pressing keys in the wrong column

my opinion about some of the questions in OP:

> Should key presses before the first hit object be ignored?
> How would 2B mapping interfere with this mod?

I think this mod should only penalize key presses that are within the hit window of another object, but don't cause a judgement. So for example, tapping during a break or tapping to fill in beats during slower sections would not hurt your score. and I think that solves the potential issue with 2B.

> Should spinners be taken into account?

I think it would be fair to have this mod do nothing during spinners. just doesn't seem important to me, and feels annoying to restrict

> How much score multiplier to give.

1x -- it restricts how you play, but doesn't change the difficulty of the map. it's like SD or PF.

**Comment by JamieWright99** [comment]
I was also looking for a post like this regarding Mania, and wanted to share something

Here's a video of a BMS program (Qwilight) that has something like this, where extra inputs are counted as added misses for your accuracy/health. In the video I press 3 keys for a single input and 2 misses are added to my count on the bottom left, with accuracy and health impacted: 

https://github.com/user-attachments/assets/d17893d1-a85d-4644-bff8-7e35e03f1eec

I prefer to play on this client because I'm still learning mania and I do not want to build bad habits around spamming, but it would be really cool for something like this to be introduced into osu!lazer
